src/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/Point.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\DBAL\Types\ValueObject;
+
+/**
+ * @since 3.1
+ *
+ * @author Sébastien Jean <sebastien.jean76@gmail.com>
+ */
+final class Point implements \Stringable
+{
+    public const POINT_REGEX = '/\((-?\d+(?:\.\d{1,6})?),\s*(-?\d+(?:\.\d{1,6})?)\)/';
+    
+    public function __construct(
```

Comment:
@coderabbitai, suggest a commitable comment on how `POINT_REGEX` can be used here to prevent incorrect object instantiation (e.g. when the precision is beyond the maximum allowed one for the said regex validation. Consider memory implications.

Reply:
Suggest how the constructor will throw on attempts to instantiate an object that will not allow the value object to resolve to a compatible value as stipulated by the regex.

Reply:
@coderabbitai Perhaps `POINT_REGEX` can be broken down into reusable components that align with your suggestion for `__construct`? Suggest a holistic refactoring.

Reply:
@seb-jean, see above, I meant something similar with [this comment](https://github.com/martin-georgiev/postgresql-for-doctrine/pull/348#discussion_r2054727444), but didn't express myself very clearly. Sorry for that.

Reply:
I made the changes :).